### DISCUSSION TITLE
Advice

### DISCUSSION THREAD

**Discussion by radiganmcse** [jstedfast/MailKit#1857] [Q&A]
Hi Everyone

I'm looking at decommissioning an on-prem Microsoft Exchange server and need a tool to send emails from some legacy apps to Microsoft Graph API. This includes printers, SQL database mail, and some custom proprietary software (we can rewrite to use Microsoft Graph API but later). 

Can Mailkit receive emails from apps/printers, connect to Graph API, and forward the messages? AKA is installed on a server and acts kind of like a mail relay.

**Comment by jstedfast** [comment]
> Can Mailkit receive emails from apps/printers

As long as the emails created by the apps/printers a valid (or even if they are not completely valid, but close enough) and those messages can be read from an IMAP or POP3 server, then yes, MailKit will be able to receive those messages.

> connect to Graph API, and forward the messages?

No, MailKit cannot connect to Graph. But you could use the [Microsoft.Graph](https://www.nuget.org/packages/Microsoft.Graph) NuGet package to forward the emails to the Graph API.

> AKA is installed on a server and acts kind of like a mail relay.

MailKit isn't a mail relay - but you could use MailKit to write a relay program.

**Reply by radiganmcse** [reply]
Good to know. Thank you!

**Reply by jstedfast** [reply]
Think of MailKit like this:

MailKit is a library that allows a programmer to talk to SMTP servers, POP3 servers, and IMAP servers. MimeKit (which is a dependency of MailKit) can parse MIME messages (and do some other stuff like S/MIME, POP, DKIM-Signatures, etc).

But neither of these are applications that run and provide some sort of service. They exist only to allow you (or someone else) to *write* a service.
